app/src/parser_impl_eth.c

```diff
@@ -0,0 +1,311 @@
+/*******************************************************************************
+ *  (c) 2018 - 2023 Zondax AG
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ********************************************************************************/
+
+#include "parser_impl_eth.h"
+#include "app_mode.h"
+#include "common/parser_common.h"
+#include "crypto.h"
+#include "eth_utils.h"
+#include "parser_txdef.h"
+#include "zxformat.h"
+#include <stdio.h>
+#include <zxmacros.h>
+
+eth_tx_t eth_tx_obj;
+
+static parser_error_t parse_field(parser_context_t *ctx, uint32_t *fieldOffset,
+                                  uint32_t *len) {
+  if (ctx->offset >= ctx->bufferLen)
+    return parser_unexpected_buffer_end;
+
+  uint32_t read = 0;
+
+  const uint8_t *data = &ctx->buffer[ctx->offset];
+
+  if (parse_rlp_item(data, ctx->bufferLen - ctx->offset, &read, len) != rlp_ok)
+    return parser_invalid_rlp_data;
+
+  *fieldOffset = ctx->offset + read;
+
+  if (*fieldOffset > ctx->bufferLen)
+    return parser_invalid_rlp_data;
+
+  ctx->offset += read + *len;
+
+  return parser_ok;
+}
+
+static parser_error_t readChainID(parser_context_t *ctx, chain_id_t *chain_id) {
```

Comment:
Check which chain IDs should be allowed here. For sure, if chain ID correspond to ETH mainnet the transaction must be rejected